interpreter/cling/lib/Utils/ASTImportSource.cpp

```diff
@@ -0,0 +1,182 @@
+        m_DeclContexts_map[importedDeclContext] = declContextFrom;
+
+        /* Not used for now.
+         * m_DeclContextsNames_map[Name.getAsString()] =
+         * std::make_pair(importedDeclContext,declContextFrom); */
+        importedDeclContext->setHasExternalVisibleStorage(true);
+
+        std::vector < NamedDecl * > declVector;
+        declVector.push_back((NamedDecl *) importedDeclContext);
+        llvm::ArrayRef < NamedDecl * > FoundDecls(declVector);
+        SetExternalVisibleDeclsForName(DC, ((NamedDecl *) importedDeclContext)->getDeclName(),
+                                       FoundDecls);
+
+        /* And also put the Decl I found from the first Interpreter
+         * in the map of the second Interpreter to have it for the future. */
+        m_DeclName_map[Name.getAsString()] = declNameI1;
+      }
+    }
+
+    bool ASTImportSource::Import(DeclContext::lookup_result lookup_result,
+                                 ASTContext &from_ASTContext,
+                                 ASTContext &to_ASTContext,
+                                 const DeclContext *DC,
+                                 DeclarationName &Name,
+                                 DeclarationName &declNameI1) {
+
+      /* Check if we found this Name in the first interpreter */
+      if (lookup_result.empty())
+        return false;
+
+      /* Prepare to import the Decl(Context)  we found in the
+       * second interpreter */
+      FileManager& fm =  m_second_Interp->getCI()->getFileManager();
+
+      /*****************************ASTImporter**********************************/
+      ASTImporter importer(to_ASTContext, fm, from_ASTContext, fm,
+        /*MinimalImport : ON*/ true);
+      /**************************************************************************/
+
+      /* If this Name we are looking for is for example a Namespace,
+       * then it is a Decl Context. */
+      if ((*lookup_result.data())->getKind() == Decl::Namespace) {
+        DeclContext *declContextFrom =
+          llvm::cast<DeclContext>(*lookup_result.data());
+        ImportDeclContext(declContextFrom, importer, Name, declNameI1, DC);
+      } else { /* If this name is just a Decl. */
+        /* Check if we have more than one results, this means that it
+         * may be an overloaded function. */
+        if (lookup_result.size() > 1) {
+          DeclContext::lookup_iterator I;
+          DeclContext::lookup_iterator E;
+          for (I = lookup_result.begin(), E = lookup_result.end(); I != E; ++I) {
+            NamedDecl *D = *I;
+            Decl *declFrom = llvm::cast<Decl>(D);
+            ImportDecl(declFrom, importer, Name, declNameI1, DC);
+          }
+        } else {
+          Decl *declFrom = llvm::cast<Decl>(*lookup_result.data());
+          ImportDecl(declFrom, importer, Name, declNameI1, DC);
+        }
+      }
+      return true;
+    }
+
+    bool ASTImportSource::FindExternalVisibleDeclsByName(const DeclContext *DC,
```

Comment:
Can you add `1` and `2` or `parent` and `child` as prefix to all `Decl` / `DeclContext`, to signal which interpreter they are from? This makes your code much more readable. E.g. is this `DC` from the child or the parent interpreter?